force-app/main/default/classes/IBMDiscoveryV1Models.cls

```diff
@@ -9515,8 +9516,17 @@ public class IBMDiscoveryV1Models {
       if (jsonMap == null) {
         return null;
       }
+      String data_updated = (String) jsonMap.remove('data_updated_serialized_name');
+      String successfully_trained = (String) jsonMap.remove('successfully_trained_serialized_name');
+      String jsonStringWithoutDates = JSON.serialize(jsonMap);
 
-      TrainingStatus ret = (TrainingStatus) super.deserialize(jsonString, jsonMap, classType);
+      TrainingStatus ret = (TrainingStatus) super.deserialize(jsonStringWithoutDates, jsonMap, classType);
+      if (String.isNotBlank(data_updated)) {
```

Comment:
Previously we were replacing empty strings with `null` to properly parse dates that weren't set. Was this causing a problem with parsing? I think I can add this date logic to the generator but want to make sure it's necessary.

Reply:
so the problem here is that the API response is messed up on the service side - it currently expects an optional `String` for these two dates. I kinda feel like changing it to `null` is correct from a developer standpoint, but from the perspective of the API, its munging the JSON (`null != ""`)

It would probably be better to have a date logic generator that handles this better (i.e. if the string is empty or null, just don't set it on the deserialization step)

Reply:
I see what you're saying. Do you know off the top of your head how blank dates will come from other services though? Like do they all come back as empty strings or are some `null`? If your solution will work for every service, I'll be happy to implement it in the generator.

Reply:
I don't know off the top of my head, but it really never should - that is a bug in the swagger spec imo for Discovery, I hope other services do not return an empty string when it should be `null` -> i'll make an issue to get it fixed though